api/utils/token/chained_validator.go

```diff
@@ -0,0 +1,28 @@
+package token
+
+import (
+	"context"
+	"errors"
+)
+
+type ChainedValidator struct{ validators []ValidatorInterface }
+
+var _ ValidatorInterface = &ChainedValidator{}
+var errNoIssuersFound = errors.New("no issuers found")
+
+func NewChainedValidator(validators ...ValidatorInterface) *ChainedValidator {
+	return &ChainedValidator{validators}
+}
+
+func (v *ChainedValidator) ValidateToken(ctx context.Context, token string) (principal TokenPrincipal, err error) {
```

Comment:
No need for names in returned values - method returns them explicitly, like `return principal, nil` or `return nil, err`

Reply:
agreed, returning a list of errors if not found :)